src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1152,6 +1146,19 @@ ucs_status_t uct_ud_ep_flush(uct_ep_h ep_h, unsigned flags,
         UCT_TL_EP_STAT_FLUSH_WAIT(&ep->super);
     }
 
+    if (ep->flags & UCT_UD_EP_FLAG_FLUSH_CANCEL) {
+        ucs_assert(flags & UCT_FLUSH_FLAG_CANCEL);
```

Comment:
1. maybe just check "if (flags & UCT_UD_EP_FLAG_FLUSH_CANCEL)" ?
2. why need to do it in the end of the function and not right after lines 1131-1134?